Content.Server/Shuttles/EntitySystems/ThrusterSystem.cs

```diff
@@ -0,0 +1,485 @@
+
+        // Used for accumulating burn if someone touches a firing thruster.
+
+        private float _accumulator;
+
+        private const float DamagePerSecond = 40f;
+
+        private const float DamagePerUpdate = 10;
+
+        private DamageSpecifier _damageSpecifier = default!;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+            SubscribeLocalEvent<ThrusterComponent, ActivateInWorldEvent>(OnActivateThruster);
+            SubscribeLocalEvent<ThrusterComponent, ComponentInit>(OnThrusterInit);
+            SubscribeLocalEvent<ThrusterComponent, ComponentShutdown>(OnThrusterShutdown);
+            SubscribeLocalEvent<ThrusterComponent, PowerChangedEvent>(OnPowerChange);
+            SubscribeLocalEvent<ThrusterComponent, AnchorStateChangedEvent>(OnAnchorChange);
+            SubscribeLocalEvent<ThrusterComponent, RotateEvent>(OnRotate);
+
+            SubscribeLocalEvent<ThrusterComponent, StartCollideEvent>(OnStartCollide);
+            SubscribeLocalEvent<ThrusterComponent, EndCollideEvent>(OnEndCollide);
+
+            _mapManager.TileChanged += OnTileChange;
+
+            var dam = IoCManager.Resolve<IPrototypeManager>().Index<DamageTypePrototype>("Heat");
+
+            _damageSpecifier =
+                new DamageSpecifier(dam, FixedPoint2.New(DamagePerUpdate));
+        }
+
+        public override void Shutdown()
+        {
+            base.Shutdown();
+            _mapManager.TileChanged -= OnTileChange;
+        }
+
+        private void OnTileChange(object? sender, TileChangedEventArgs e)
+        {
+            // If the old tile was space but the new one isn't then disable all adjacent thrusters
+            if (e.NewTile.IsSpace() || !e.OldTile.IsSpace()) return;
+
+            var tilePos = e.NewTile.GridIndices;
+
+            for (var x = -1; x <= 1; x++)
+            {
+                for (var y = -1; y <= 1; y++)
+                {
+                    if (x != 0 && y != 0) continue;
+
+                    var checkPos = tilePos + new Vector2i(x, y);
+
+                    foreach (var ent in _mapManager.GetGrid(e.NewTile.GridIndex).GetAnchoredEntities(checkPos))
+                    {
+                        if (!EntityManager.TryGetComponent(ent, out ThrusterComponent? thruster) || thruster.Type == ThrusterType.Angular) continue;
+
+                        // Work out if the thruster is facing this direction
+                        var direction = EntityManager.GetComponent<TransformComponent>(ent).LocalRotation.ToWorldVec();
+
+                        if (new Vector2i((int) direction.X, (int) direction.Y) != new Vector2i(x, y)) continue;
+
+                        DisableThruster(ent, thruster);
+                    }
+                }
+            }
+        }
+
+        private void OnActivateThruster(EntityUid uid, ThrusterComponent component, ActivateInWorldEvent args)
+        {
+            component.EnabledVV ^= true;
```

Comment:
i dont like that this is being set in non-VV places

Reply:
TBH I couldn't think of a name for it because I have 2 enabled properties, I could call the other one "On" and just call this enabled.